File: crates/codegraph/tests/fixtures/multi_lang/ts_app/src/handlers.ts
import { authenticate } from "./auth";
import { newUser, User } from "./user";

export function login(name: string): User | null {
  const u = newUser(1, name);
  return authenticate(u) ? u : null;
}

export function whoami(u: User): string {
  return u.name;
}

File: crates/codegraph/tests/fixtures/multi_lang/rust_app/src/handlers.rs
use crate::auth::authenticate;
use crate::{new_user, User};

pub fn login(name: &str) -> Option<User> {
    let u = new_user(1, name);
    if authenticate(&u) {
        Some(u)
    } else {
        None
    }
}

pub fn whoami(u: &User) -> String {
    u.name.clone()
}
